numpy/ma/core.py

```diff
@@ -7683,6 +7676,10 @@ def asanyarray(a, dtype=None):
     <class 'numpy.ma.core.MaskedArray'>
 
     """
+    # workaround for #8666, to preserve identity. Ideally the bottom line
```

Comment:
@mhkv: Not enough. I think you also need to check that all of the other arguments are their default value